extensions/amp-access/amp-access-spec.md

```diff
@@ -319,10 +326,20 @@ Notice the use of a URL hash parameter “status”. The value is either “true
 An integration with *amp-analytics* is under development and can be tracked on [Issue #1556][10]. This document will be updated when more details on the integration are available.
 
 #CORS Origin Security
-CORS endpoints can be (and recommended) to be only allowed to the following origins:
+
+Authorization and Pingback endpoints are CORS endpoints and they must implement CORS security protocol. This protocol consists of two components: CORS origin and source origin restrictions.
+
+CORS endpoints receive requesting origin via Access-Control-Allow-Origin HTTP header. This header has to be restricted to only allow the following origins:
  - *.ampproject.org
  - The Publisher’s own origins
 
+Source origin restrictions has to be implemented by requiring "__amp_source_origin" URL parameter to be within a set of the Publisher's own origins. The "__amp_source_origin" parameter is passed from AMP Runtime in all fetch requests and contains the source origin, e.g. "https://publisher1.com". The resulting response has to also contain the following headers:
+- `AMP-Access-Control-Allow-Source-Origin: https://publisher1.com`. Here "https://publisher1.com" indicates the source origin that is allowed to read the authorization response.
+- `Access-Control-Expose-Headers: AMP-Access-Control-Allow-Source-Origin`. This header simply allows CORS response to contain the first header.
```

Comment:
It would be useful to include Access-Control-Allow-Origin in the exposed header list. That would make it possible to check in the client code if the publisher really has a proper whitelist for origins instead of '*', so that we can prevent them from creating an unsafe API endpoint.